### PR TITLE
Slot machine

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/field/slot_machine.c [hunk 1]
```diff
@@ -1339,6 +1339,69 @@ bool8 sub_8102EA4(struct Task *task)
...
+    if (eSlotMachine->unk0A == 0 && (eSlotMachine->unk04 == 0 || eSlotMachine->unk06 == 0 || !gUnknown_083ECB40[task->data[15]]()))
+    {
+        eSlotMachine->unk06 = 0;
+        gUnknown_083ECB4C[task->data[15]]();
+    }
+    task->data[1] = eSlotMachine->unk2E[task->data[15]];
+    return TRUE;
+}
+
+extern const u16 gUnknown_083ECB58[5]; // don't move this
```
Comment (ProjectRevoTPP) [inline src/field/slot_machine.c]: does the ordering matter?

Comment (PikalaxALT) [reply src/field/slot_machine.c]: I plan to nuke the memcpy call later by replacing it with the commented line below

File: src/field/slot_machine.c [hunk 2]
```diff
@@ -20,7 +20,7 @@
 struct UnkStruct1 {
     /*0x00*/ u8 unk00;
     /*0x01*/ u8 unk01;
-    /*0x02*/ s16 pikaPower;
+    /*0x02*/ s16 unk02;
```
Comment (ProjectRevoTPP) [inline src/field/slot_machine.c]: regression?

Comment (PikalaxALT) [reply src/field/slot_machine.c]: mistake corrected

**Pull request merged**